### PR TITLE
virtcontainers: allow specifying nydus-overlayfs binary by path

### PR DESCRIPTION
**Pull request by katexochen**
...or by using a binary with additional suffix.

This allows having multiple verisons of nydus-overlayfs installed on the host, telling nydus-snapshotter which one to use while still detecting Nydus is used.

### CODE REVIEW

**Review by ChengyuZhu6 — APPROVED** [review]
I understand the purpose of your implementation, and overall, the code looks good to me.

**Comment by katexochen** [comment]
Thanks for your review @ChengyuZhu6! Could you also take a look at https://github.com/containerd/nydus-snapshotter/pull/615?

**Comment by katexochen** [comment]
Fixed the Go formatting of one file.

**Comment by ChengyuZhu6** [comment]
> Thanks for your review @ChengyuZhu6! Could you also take a look at [containerd/nydus-snapshotter#615](https://github.com/containerd/nydus-snapshotter/pull/615)?

I'll take a look.

**Comment by katexochen** [comment]
nydus-snapshotter PR was merged.

**Comment by stevenhorsman** [comment]
@katexochen - do we need to wait for a nydus-snapshotter release and then bump the version in https://github.com/kata-containers/kata-containers/blob/f31839af632e6c70f0ac7f5fb6f689e7543981a1/versions.yaml#L330 for this PR, or is it ready before that?

**Comment by katexochen** [comment]
> @katexochen - do we need to wait for a nydus-snapshotter release and then bump the version in
> 
> https://github.com/kata-containers/kata-containers/blob/f31839af632e6c70f0ac7f5fb6f689e7543981a1/versions.yaml#L330
> 
> for this PR, or is it ready before that?

IDK, @fidencio wanted me to wait until the nydus-snapshotter PR is merged. As we aren't using this feature in kata right now I'm not sure whether we should bump/wait for release. Edgeless is using this downstream.

**Comment by ChengyuZhu6** [comment]
> > @katexochen - do we need to wait for a nydus-snapshotter release and then bump the version in
> > https://github.com/kata-containers/kata-containers/blob/f31839af632e6c70f0ac7f5fb6f689e7543981a1/versions.yaml#L330
> > 
> > for this PR, or is it ready before that?
> 
> IDK, @fidencio wanted me to wait until the nydus-snapshotter PR is merged. As we aren't using this feature in kata right now I'm not sure whether we should bump/wait for release. Edgeless is using this downstream.

Let me sync with the repo maintainers.

**Comment by katexochen** [comment]
ping @ChengyuZhu6 any update on this? Would be nice if we could get this into the next release.

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks @katexochen!

File: src/runtime/virtcontainers/mount_linux.go
Comment (sprt) [inline src/runtime/virtcontainers/mount_linux.go:187]: @katexochen Should it be this instead like elsewhere?
```suggestion
			if IsNydusRootFSType(c.rootFs.Type) {
```

Comment (katexochen) [reply src/runtime/virtcontainers/mount_linux.go:187]: Yes, looks like it. Did you open a fix for this somewhere already?

Comment (sprt) [reply src/runtime/virtcontainers/mount_linux.go:187]: I haven't, just noticed this randomly!

**Pull request merged**
